### PR TITLE
Add details to 'about_Modules' topic

### PR DESCRIPTION
**Pull request by guillermooo**
# PR Summary

Reorganize paragraphs and try to provide details why modules can help.

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

**Comment by guillermooo** [comment]
@michaeltlombardi Before updating the other files, please check if this looks OK to you.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md [hunk 1]
```diff
@@ -13,29 +13,31 @@ Explains how to install, import, and use PowerShell modules.
...
-module.
-
-Modules can be created as compiled .NET assemblies written in C#, or
-script-based modules written in PowerShell. This topic explains how to use
-PowerShell modules. For information about how to write PowerShell modules, see
-[Writing a PowerShell Module][02].
+PowerShell is both a command shell and a scripting language. Commands in
+PowerShell are implemented as scripts, functions, or cmdlets. The language
+includes [keywords][07], which provide the structure and logic of processing,
+and other resources, such as variables, providers, aliases, and more.
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md:19]: The prefixing phrase "such as" indicates that the list is non-exhaustive.

```suggestion
and other resources, such as variables, providers, aliases.
```

**Review by michaeltlombardi — COMMENTED** [review]
@guillermooo - I'm sorry for the late reply to your PR, it looks like my prior review didn't submit correctly.

What you have here looks good, I like the restructuring and clarifications. I have a few minor recommendations, but nothing major. I think you can continue this work and I'll make sure to review it promptly.

One thing to keep in mind is that we need to limit the line length in about topics to 80 characters or fewer (see [the markdownlint config][aa]). You can use the [Reflow Markdown VS Code extension][ab] to make that task easier. I see that our contributor guide doesn't point this out, so I'll update that shortly.

[aa]: https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/.markdownlint.yaml
[ab]: https://marketplace.visualstudio.com/items?itemName=marvhen.reflow-markdown

**Review by sdwheeler — APPROVED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md [hunk 2]
```diff
@@ -13,29 +13,31 @@ Explains how to install, import, and use PowerShell modules.
...
+installed module the first time you use a command from the module. You can configure
+automatic module loading behavior using the variable `$PSModuleAutoloadingPreference`.
+For more information, see [about_Preference_Variables][08].
+
+You can also manually unload or reload modules during a PowerShell session. To unload a
+module, use the `Remove-Module` cmdlet. To load or reload a module, use `Import-Module`.
+
+PowerShell comes with a base set of modules. You or third parties can create
+new PowerShell commands or other resources, and publish them as modules that
+users can install as needed.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md]: This is already linked in a more appropriate location in the doc.

**Pull request merged**